Adds display unit preference

Anish-Chanda opened the pull request:
Adding unit display preferences for Temperature, Networking, and Disk throughput. fixes #402

henrygd commented:
Cool, thanks. I should have a beta version of the next release out in the next few days and I'll follow up on the open PRs afterwards.

Anish-Chanda commented:
changes:
- updated settings UI to add new options
- updated `stores.ts` and `internal/users.go` to store the user's unit preferences 
- new conversion helpers in `utils.ts`
- updated area container (we check which charts to add units to from their name) and temperature charts to support units.

Added options to select unit preferences under general settings
![image](https://github.com/user-attachments/assets/f3265abd-d06f-4d8c-8cd5-f964804b742e)

screenshots with Fahrenheit for temp, bps for networking and MB/s for disk throughput.
![image](https://github.com/user-attachments/assets/dd72bd2c-8c32-4eed-80c6-b908fe692dab)
![image](https://github.com/user-attachments/assets/d236bef5-778b-4e7e-b9b5-e3297b155c29)

henrygd commented:
Thanks, I do want to change the data coming in from the agent to use bytes at some point, rather than MB, which will make these conversions more accurate.

Anish-Chanda commented:
@henrygd first we could make a change in the hub that converts existing `dr`, `dw` etc in system stats from MB to bytes, and based on the agents version we either write it to db directly or for older agents we convert it to bytes. in the subsequent release for the agent we could start pushing bytes instead of MB. there will still be a bit of inaccuracy in the historic data but once the agents are updated we should have accurate data for these charts. this would make this change the least disruptive. I'm also open to alternative approaches or discussion if anyone has other ideas.

henrygd commented:
That would be a good way to do it. Or add an extra property like `bytes: true`.

0.12.0 beta is slightly delayed so I can get another feature in, but I'll be done very soon. We may be able to get this in the final 0.12.0 release. There might be conflicts when I merge my dev branch but don't worry about that.

henrygd commented:
Merged, thanks! I expanded it a bit so the choices are bits and bytes, and the bytes option will show the whole range (KB, MB, etc).
